packages/vitest/src/node/plugins/index.ts

```diff
@@ -99,6 +92,16 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
         )
         options.api = resolveApiConfig(options)
 
+        // we cannot replace them, because esbuild will throw an error,
+        // so we are removing them to allow reassigning
+        const define = viteConfig.define
+
+        if (define) {
+          delete define['process.env.NODE_ENV']
+          delete define['global.process.env.NODE_ENV']
+          delete define['globalThis.process.env.NODE_ENV']
+        }
```

Comment:
Non-blocking comment: Are we deleting these entries in case the user added them by hand, no? Should we instead filter them out where they are causing issues? (with a patch in Vite if necessary). Or maybe I'm missing some context, and the comment could be expanded a bit.

Reply:
Yeah I kinda agree not to delete them. If user define them and get the error, it's more like their responsiblity to either fix it or find a workaround

Reply:
> Yeah I kinda agree not to delete them. If user define them and get the error, it's more like their responsiblity to either fix it or find a workaround

Then we shouldn't add them in Vite